crates/proto/src/xfer/dnssec_dns_handle/nsec3_validation.rs

```diff
@@ -1,3 +1,72 @@
+//! Throughout this module several NSEC3 specific terms are used.
+//!
+//! "closest_encloser" - a name that is a parent / grandparent /
+//!     great-grandparent, etc. of a `query_name`, that DOES exist.
+//!
+//! "next_closer" - a name that is one level deeper in the hierarchy from
+//!     `closest_encloser`, that DOES NOT exist in the zone.
+//!
+//! "wildcard of closest encloser" - a name in a form of `*.closest_encloser`.
+//!     If it exists then `query_name` would be serviced by the wildcard.
+//!
+//! "covering NSEC3 record" - NSEC3 record name has a hash (`hashed_owner_name`),
+//!     and inside the record data there's `next_hashed_owner_name`.
+//!     If the hash of `query_name` fits between the two hashes then the record
+//!     "covers" `query_name`
+//!
+//! "own NSEC3 record" - exists for *existing* names only.
+//!     The `hashed_owner_name` would match one of the names exactly.
+//!
+//! In general:
+//! * if a name exists we would expect to see its "own" NSEC3 record,
+//! * if the name doesn't exist we would expect to see a "covering" NSEC3 record.
```

Comment:
Do you think is necessary to mention how wildcards fit here? You can compute a cover to prove that no record exactly matches QNAME but you still need to prove that there is no record owned by a wildcard name that matches QNAME as well.

Reply:
No, this is discussed further down the line in the code, and I want to keep this docs segment short. I also don't discuss record types here, so that people don't get overwhelmed. NSEC3 is confusing enough as is.